### DISCUSSION TITLE
Linktree Importer

### DISCUSSION THREAD

**Discussion by benkaiser** [JulianPrieber/littlelink-custom#712] [Feature requests]
@JulianPrieber has there been any interest in an automatic linktree importer? Allowing users to input a linktree URL and bring across all their links automatically?

I'd be open to creating a pull request adding support for this feature.

**Comment by JulianPrieber** [comment]
https://www.reddit.com/r/selfhosted/comments/i1xaht/comment/ikui3z1/?utm_source=reddit&utm_medium=web2x&context=3

**Comment by JulianPrieber** [comment]
Id suggest working out a proof of concept/third party tool that exports linktree links as such: [all-links.json](https://github.com/LinkStackOrg/LinkStack/files/14112249/all-links.json).
LinkStack can then import this as native links.

**Comment by benkaiser** [comment]
@JulianPrieber I actually have an old npm package I can update: https://www.npmjs.com/package/linktree-scraper

We could snap to an agreed upon format to normalize too, then the package can return that normalized format even if the underlying data structure from linktree changes (obviously this package would need to be bumped whenever linktree break the contract, since it's an internal representation).

**Comment by benkaiser** [comment]
I've updated the library to perform some of the normalization: https://www.npmjs.com/package/linktree-scraper

Would it be simple enough if we re-implemented [the core logic](https://github.com/benkaiser/linktree-scraper/blob/ea471c1e63f4c6d1a77cb6e9a1546f367e128862/src/index.ts#L24C1-L59C2) inside the PHP in the codebase of LinkStack?

**Reply by JulianPrieber** [reply]
https://gist.github.com/JulianPrieber/dfb19d1b786ffe06eeaf0db0796ea45c

**Reply by benkaiser** [reply]
Wow, seems like you've got it handled!

**Reply by JulianPrieber** [reply]
I'm uncertain about integrating a crawler into the application. I like to keep things self-contained, as relying on external sources functionality could be affected by any changes or decisions made by Linktree.
Too bad, we don't have a sort of third party plugin system.

I might just turn this into a tool on the website, then just allow users to import it from there.

Maybe something open source would be better? But then again, that might be a bit overkill for a tool like this.
